statement.go

```diff
@@ -689,7 +689,7 @@ func (statement *Statement) Top(limit int) *Statement {
 
 // Limit generate LIMIT start, limit statement
 func (statement *Statement) Limit(limit int, start ...int) *Statement {
-	statement.LimitN = limit
+	statement.LimitN = &limit
```

Comment:
I think `statement.LimitN` here will never get a nil val here since you've already passed an non-pointer type value within param `limit`. Furthermore, you should also write some cases to test your code.

Reply:
`if limit > 0`

Reply:
And maybe we still need an extra param to indicate that value "0" in param `limit` is specified intentionally or not(considering using a bool param?).

Reply:
a critical bug, it did delete all my table data, but i just want to delete none

Reply:
@Frank-Hust I will accept your PR, could you please resolve the conflict and add some tests for it?